dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -725,7 +725,12 @@ public Workflow publish(@ApiParam(hidden = true) @Auth User user, @ApiParam(valu
                 .workflowsSharedWithUser(user).entrySet().stream()
                 .map(e -> {
                     final List<Workflow> workflows = e.getValue().stream().map(path -> {
-                        return workflowDAO.findByPath(path, false);
+                        final Workflow workflow = workflowDAO.findByPath(path, false);
```

Comment:
Seems a bit slow to retrieve each workflow individually from the DB. I think we have an endpoint for users to retrieve all their workflows as a group, can probably re-use that, create a map and then use as a filter.

Reply:
Unless I'm misunderstanding, the problem is these are not the user's workflows in the sense that the user would be present in Workflow.getUsers(). These are workflows that have been shared via SAM and hence the relationship is not captured in the Dockstore database tables.

I also think this wouldn't typically be a very large number of workflows.

Or are you suggesting dynamically doing a CriteriaQuery like `io.dockstore.webservice.jdbi.EntryDAO#findAllPublished(java.lang.String, java.lang.Integer, java.lang.String, java.lang.String, java.lang.String)`?

Reply:
I think the latter.

My understanding is that `permissionsInterface.workflowsSharedWithUser(user)` is grabbing the small number of workflows that are in sam, but then in the map step, you are filtering out the workflows that are not a part of a user's workflows due to Workflow.getUsers() by retrieving each one from the database and checking. 

There's probably a better way of doing that in one query using JPA, but I think we may be talking premature optimization so probably best to just lock down behaviour with a test, put a `//TODO` to indicate the idea of how to make it faster and move on.